src/ucp/proto/proto_perf.c

```diff
@@ -176,6 +175,21 @@ ucp_proto_perf_node_update_factors(ucp_proto_perf_node_t *perf_node,
     }
 }
 
+static void
+ucp_proto_perf_segment_update_factor(ucp_proto_perf_segment_t *seg,
+                                     ucp_proto_perf_factor_id_t factor_id,
+                                     ucs_linear_func_t perf_factor)
+{
+    seg->perf_factors[factor_id] = perf_factor;
+    if (ucs_linear_func_is_zero(perf_factor, UCP_PROTO_PERF_EPSILON)) {
+        return;
+    }
+
+    ucp_proto_perf_node_update_data(seg->node,
+                                    ucp_proto_perf_factor_names[factor_id],
+                                    perf_factor);
```

Comment:
this block seems very similar to ucp_proto_perf_node_update_factors body, can we unite common code?